### PR TITLE
Deprecation messages and guides

### PR DESCRIPTION
**Pull request by mxriverlynn**
### Description

This PR continues the work to [improve the deprecation architecture and messages for PVC](https://github.com/github/primer/discussions/1515), making the following changes:

* Documentation for deprecations has bee moved to `docs/contributors/deprecations.md`

* A "guide" setting has been added to the `lib/primer/deprecations.yml` configuration, allowing a component to specify an update/migration guide that will be shown with the deprecation message. 

* The message that is displayed for deprecations has been significantly modified. Now, instead of a singular message regardless of the component having a replacement or guide, there are several deprecation states that change the details of the message. 

* A component can now be marked as deprecated using code or using a `.yml` file. See `docs/contributors/deprecations.md` for more information

* The `DeprecatedComponentsHelper` at `lib/primer/view_components/linters/helpers/deprecated_components_helpers.rb` has been updated to pull information from the new deprecations configuration, rather than the `static/statuses.json` file. 

* Tests have been added to cover the valid and invalid configuration scenarios for deprecations, ensuring all configurations are valid. See the "Valid and Invalid Deprecation Configurations" below, for information on what is and is not valid. There are also tests to cover the new messages that are displayed for deprecations.

* The `component_status_migrator.thor` script has been updated to correct the new `deprecations.yml` file, instead of looking for the old deprecations entries. It has also been updated to ignore non-zero exit codes from running rubocop, which were preventing it from completing. And a message is displayed at the end of this script, instructing developers to update the `deprecations.yml` file.

### Integration

> Does this change require any updates to code in production?

No immediate changes are needed. This 

### Merge checklist

- [x] Added/updated tests
- [x] Added/updated documentation
- [ ] Added/updated previews

### CODE REVIEW

**Review by jonrohan — COMMENTED** [review]
This is great! I wonder if it's worth adding what you've written in the PR description and yml file about the process to the `docs/contributors/` folder?

File: test/lib/deprecation_messages_test.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  end
+
+  def test_replacement_correctable_guide
+    msg = deprecation_message({
+                                replacement: @replacement,
+                                autocorrect: true,
+                                guide: @guide
+                              })
+
+    assert_equal "'#{@component}' has been deprecated. Please update your code to use '#{@replacement}' or use rubocopy's auto-correct option to do it for you. See #{@guide} for more information.", msg
```
Comment (camertron) [inline test/lib/deprecation_messages_test.rb:36]:
```suggestion
    assert_equal "'#{@component}' has been deprecated. Please update your code to use '#{@replacement}' or use rubocop's auto-correct option to do it for you. See #{@guide} for more information.", msg
```

File: test/lib/deprecation_messages_test.rb [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  end
+
+  def test_replacement_correctable_no_guide
+    msg = deprecation_message({
+                                replacement: @replacement,
+                                autocorrect: true
+                              })
+
+    assert_equal "'#{@component}' has been deprecated. Please update your code to use '#{@replacement}' or use rubocopy's auto-correct option to do it for you.", msg
```
Comment (camertron) [inline test/lib/deprecation_messages_test.rb:45]:
```suggestion
    assert_equal "'#{@component}' has been deprecated. Please update your code to use '#{@replacement}' or use rubocop's auto-correct option to do it for you.", msg
```

**Review by camertron — COMMENTED** [review]
Wow awesome! Just a few spelling things 🎉

**Review by jonrohan — APPROVED** [review]
❤️

**Pull request merged**